roles/gather/tasks/main.yaml

```diff
@@ -19,15 +19,50 @@
             )
         }}
 
+- name: Add gathered system data structure under 'invnetory_name' key
+  ansible.builtin.set_fact:
+    "{{ inventory_hostname }}":
+      all_gathered_resources: >-
+        {{
+          (all_gathered_resources | default({}))
+          | combine(
+              {'device_info': (system_resources | default({})) | combine({'ansible_network_os': ansible_network_os | default('')})}
+            )
+        }}
+    
+- name: Initialize network report dictionary
+  ansible.builtin.set_fact:
+    network_reports: {}
+  run_once: true
+  delegate_to: localhost
+  
+- name: Aggregate gathered data from each host
```

Comment:
Can we also update the README for the respective roles with latest multi device examples